GeneratorInterface/RivetInterface/plugins/HTXSRivetProducer.cc

```diff
@@ -0,0 +1,257 @@
+/* \class HTXSRivetProducer
+ *
+ * \author David Sperka, University of Florida
+ *
+ * $Id: HTXSRivetProducer.cc,v 1.1 2016/09/27 13:07:29 dsperka Exp $
+ *
+ */
+
+#include "FWCore/Framework/interface/EDProducer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Framework/interface/Run.h"
+
+#include "SimDataFormats/GeneratorProducts/interface/HepMCProduct.h"
+#include "SimDataFormats/GeneratorProducts/interface/LHERunInfoProduct.h"
+
+#include "Rivet/AnalysisHandler.hh"
+#include "GeneratorInterface/RivetInterface/src/HiggsTemplateCrossSections.cc"
+#include "SimDataFormats/HTXS/interface/HiggsTemplateCrossSections.h"
+
+#include <vector>
+#include <cstdio>
+#include <cstring>
+
+
+using namespace Rivet;
+using namespace edm;
+using namespace std;
+
+class HTXSRivetProducer : public edm::EDProducer {
```

Comment:
hi @perrozzi - back to the code review 

please use a non-legacy producer type.

Reply:
sorry I would need an example

Reply:
there should be many around - eg, a stream::producer here
https://github.com/cms-sw/cmssw/blob/master/RecoMuon/StandAloneMuonProducer/src/StandAloneMuonProducer.h